numpy/lib/tests/test_format.py

```diff
@@ -963,3 +963,33 @@ def test_unicode_field_names():
     with open(fname, 'wb') as f:
         with assert_warns(UserWarning):
             format.write_array(f, arr, version=None)
+
+
+@pytest.mark.parametrize('dt, fail', [
+    (np.dtype({'names': ['a', 'b'], 'formats':  [float, np.dtype('S3',
+                 metadata={'some': 'stuff'})]}), True),
+    (np.dtype(int, metadata={'some': 'stuff'}), False),
+    (np.dtype(int, (2,), metadata={'some': 'stuff'}), False),
```

Comment:
This isn't doing anything reasonable - you've written `align=(2,)` I think.

You want `np.dtype((int, (2,)), metadata=...)`, with extra parens.

Of course, as you remark above that metadata gets dropped on array creation - so you probably want

`np.dtype([('subarray', (int, (2,))], metadata=...)`

Reply:
adopted